### PR TITLE
BackupBrowser: Update the header style of the backup browser

### PR DESCRIPTION
**Pull request by jwebbdev**
Related to p1HpG7-nNV-p2#comment-65834

## Proposed Changes

* Updates the header to keep checkboxes always visible and reduces the number of buttons on the screen to just the headers

## Testing Instructions

* Test locally or via a live branch.  If using a live branch, add ?flags=jetpack/backup-granular to the URL
* Visit the Backup Browser page for a site
* Visual check vs. the linked comment / figma
* Ensure the buttons still function as they do in production

* Before
Screenshot 2023-09-07 at 12 30 28 PM
Screenshot 2023-09-07 at 12 30 35 PM

* After
Screenshot 2023-09-07 at 12 29 23 PM
Screenshot 2023-09-07 at 12 29 38 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/backup/backup-contents-page/file-browser/file-browser-header.tsx [hunk 1]
```diff
@@ -59,57 +46,33 @@ const FileBrowserHeader: FunctionComponent< FileBrowserHeaderProps > = ( {
...
-						borderless
-						compact
 					>
-						<Icon icon={ close } />
+						{ translate( 'Restore selected files' ) }
 					</Button>
 				</div>
 			) }
+			<div className="file-browser-header__selecting">
+				<FormCheckbox
```
Comment (Initsogar) [inline client/my-sites/backup/backup-contents-page/file-browser/file-browser-header.tsx:73]: I think for now, we should show the checkboxes only if the feature flag is enabled.

File: client/my-sites/backup/backup-contents-page/file-browser/file-browser-header.tsx [hunk 2]
```diff
@@ -59,57 +46,33 @@ const FileBrowserHeader: FunctionComponent< FileBrowserHeaderProps > = ( {
...
-						{ translate( 'Download files' ) }
+			{ browserCheckList.totalItems > 0 && (
+				<div className="file-browser-header__action-buttons">
+					<Button className="file-browser-header__download-button" onClick={ onDownloadClick }>
+						{ translate( 'Download selected files' ) }
 					</Button>
 					<Button
 						className="file-browser-header__restore-button"
 						onClick={ onRestoreClick }
+						disabled={ isRestoreDisabled }
```
Comment (Initsogar) [inline client/my-sites/backup/backup-contents-page/file-browser/file-browser-header.tsx:65]: We might want to hide the restore button if the site is a multisite, similar to [ActionButtons behavior](https://github.com/Automattic/wp-calypso/blob/5a881f0ddd815e1346e55f171572a8fd93833160/client/components/jetpack/daily-backup-status/action-buttons.jsx#L109).
This also applies to the single restore button. We can address that in a separate PR.

**Comment by Initsogar** [comment]
In general, it looks good and cleaner to me. We just need to hide the checkbox if the feature flag is not enabled and, I think we should merge https://github.com/Automattic/wp-calypso/pull/81339 before proceeding with this PR.

**Review by Initsogar — APPROVED** [review]
Looks good to me 👍🏽

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8880892

Thank you @jwebbdev for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**